### ISSUE TITLE
minishift registry-route addon seems to block docker push to minishift registry

### ISSUE THREAD

**Issue by cunningt** [minishift/minishift#1933]
### General information

  * Minishift version: 
minishift v1.11.0+4459917, but reproduces on 1.10 as well
  * OS: Linux / macOS / Windows
Mac OS X 10.13.2
  * Hypervisor: KVM / xhyve / Hyper-V / VirtualBox
xhyve

tcunning@tcunning-OSX:~/src/cunningt/prometheus-asb-installer/fuseprometheus-apb$  docker version
Client:
 Version:	17.12.0-ce
 API version:	1.24 (downgraded from 1.35)
 Go version:	go1.9.2
 Git commit:	c97c6d6
 Built:	Wed Dec 27 20:03:51 2017
 OS/Arch:	darwin/amd64

Server:
 Engine:
  Version:	1.12.6
  API version:	1.24 (minimum version )
  Go version:	go1.8.3
  Git commit:	85d7426/1.12.6
  Built:	Tue Oct 24 15:40:21 2017
  OS/Arch:	linux/amd64
  Experimental:	false

### Steps to reproduce

tcunning@tcunning-OSX:~/src/cunningt/prometheus-asb-installer/fuseprometheus-apb$ minishift addons list
- admin-user     : disabled   P(0)
- anyuid         : disabled   P(0)
- registry-route : disabled   P(0)
- xpaas          : disabled   P(0)

```
tcunning@tcunning-OSX:~/src/cunningt/prometheus-asb-installer/fuseprometheus-apb$ minishift addons enable registry-route
Add-on 'registry-route' enabled
tcunning@tcunning-OSX:~/src/cunningt/prometheus-asb-installer/fuseprometheus-apb$ minishift start --openshift-version=v3.7.0 --iso-url centos --memory 4GB
-- Starting profile 'minishift'
-- Checking if requested hypervisor 'xhyve' is supported on this platform ... OK
-- Checking if xhyve driver is installed ... 
   Driver is available at /usr/local/bin/docker-machine-driver-xhyve
   Checking for setuid bit ... OK
-- Checking the ISO URL ... OK
-- Starting local OpenShift cluster using 'xhyve' hypervisor ...
-- Minishift VM will be configured with ...
   Memory:    4 GB
   vCPUs :    2
   Disk size: 20 GB

   Downloading ISO 'https://github.com/minishift/minishift-centos-iso/releases/download/v1.4.0/minishift-centos7.iso'
 324.00 MiB / 324.00 MiB [====================================================] 100.00% 0s
-- Starting Minishift VM ............... OK
-- Checking for IP address ... OK
-- Checking if external host is reachable from the Minishift VM ... 
   Pinging 8.8.8.8 ... OK
-- Checking HTTP connectivity from the VM ... 
   Retrieving http://minishift.io/index.html ... OK
-- Checking if persistent storage volume is mounted ... OK
-- Checking available disk space ... 1% used OK
-- Downloading OpenShift binary 'oc' version 'v3.7.0'
 37.60 MiB / 38.18 MiB [======================================================]  98.49% 0s 38.18 MiB / 38.18 MiB [======================================================] 100.00% 0sOK
-- OpenShift cluster will be configured with ...
   Version: v3.7.0
-- Checking `oc` support for startup flags ... 
   host-volumes-dir ... OK
   version ... OK
   routing-suffix ... OK
   host-config-dir ... OK
   host-data-dir ... OK
   host-pv-dir ... OK
Starting OpenShift using openshift/origin:v3.7.0 ...
Pulling image openshift/origin:v3.7.0
Pulled 1/4 layers, 26% complete
Pulled 2/4 layers, 74% complete
Pulled 3/4 layers, 80% complete
Pulled 4/4 layers, 100% complete
Extracting
Image pull complete
OpenShift server started.

The server is accessible via web console at:
    https://192.168.64.46:8443

You are logged in as:
    User:     developer
    Password: <any value>

To login as administrator:
    oc login -u system:admin

-- Applying addon 'registry-route':........
Add-on 'registry-route' created docker-registry route. Please run following commands to login to the OpenShift docker registry:
$ eval $(minishift docker-env)
$ docker login -u developer -p `oc whoami -t` docker-registry-default.192.168.64.46.nip.io:443

tcunning@tcunning-OSX:~/src/cunningt/prometheus-asb-installer/fuseprometheus-apb$ eval $(minishift docker-env)
tcunning@tcunning-OSX:~/src/cunningt/prometheus-asb-installer/fuseprometheus-apb$ docker login -u developer -p `oc whoami -t` docker-registry-default.192.168.64.46.nip.io:443
WARNING! Using --password via the CLI is insecure. Use --password-stdin.
Login Succeeded
tcunning@tcunning-OSX:~/src/cunningt/prometheus-asb-installer/fuseprometheus-apb$ docker pull nginx:alpine
Trying to pull repository docker.io/library/nginx ... 
alpine: Pulling from docker.io/library/nginx
550fe1bea624: Pull complete 
d421ba34525b: Pull complete 
fdcbcb327323: Pull complete 
bfbcec2fc4d5: Pull complete 
Digest: sha256:c8ff0187cc75e1f5002c7ca9841cb191d33c4080f38140b9d6f07902ababbe66
tcunning@tcunning-OSX:~/src/cunningt/prometheus-asb-installer/fuseprometheus-apb$ docker tag nginx:alpine $(minishift openshift registry)/myproject/my-app
tcunning@tcunning-OSX:~/src/cunningt/prometheus-asb-installer/fuseprometheus-apb$ docker push $(minishift openshift registry)/myproject/my-app
The push refers to a repository [docker-registry-default.192.168.64.46.nip.io:443/myproject/my-app]
f93c2b24cb18: Pushing  4.608kB
343bb8320f2b: Retrying in 1 second 
7066df57739c: Pushing  12.83MB/12.83MB
d39d92664027: Pushing  3.988MB/3.988MB
EOF
tcunning@tcunning-OSX:~/src/cunningt/prometheus-asb-installer/fuseprometheus-apb$ echo $?
1
```

### Expected

Expected result would be that the push would be successful.      

### Actual

Every push I try with the registry-route addon enable ends up the same - a lot of retries and an eventual failure.

**Comment by gbraad** [comment]
@praveenkumar I think this is something we have seen before, right? I also think we added documentation about this... or have an issue.

**Comment by praveenkumar** [comment]
@cunningt I did try it on mac with centos images and not able to reproduce it. Can you please try it again and see if works?

**Comment by praveenkumar** [comment]
@cunningt I am able to reproduce it with 3.7.1, will look into it more and get back to you.

**Comment by amitkrout** [comment]
Now its reproducible on minishift master as v3.7.1 is default openshift version

**Comment by LalatenduMohanty** [comment]
@praveenkumar Does https://docs.openshift.org/latest/minishift/openshift/openshift-docker-registry.html was working fine. I means user who do not use the add-on was it working for them?

**Comment by LalatenduMohanty** [comment]
@praveenkumar I tested the steps to reproduce and the below steps seems wrong i.e. `minishift openshift regsistry` should not be used with the registry add-on isn't it?

```
$ docker tag nginx:alpine $(minishift openshift registry)/myproject/my-app

$ docker push $(minishift openshift registry)/myproject/my-app
The push refers to a repository [172.30.1.1:5000/myproject/my-app]
f93c2b24cb18: Preparing
343bb8320f2b: Preparing
7066df57739c: Preparing
d39d92664027: Preparing
unauthorized: authentication required
```

**Comment by praveenkumar** [comment]
@LalatenduMohanty in the documents we have https://docs.openshift.org/latest/minishift/openshift/openshift-docker-registry.html#login-to-registry following listed.

```
 $ docker login -u developer -p $(oc whoami -t) $(minishift openshift registry)
```

so if a user doesn't use addon then the output of `minishift openshift registry` would be `172.30.1.1:5000` but in case of addon enabled then it would be `docker-registry<IP>.nix.io:443` .

**Comment by LalatenduMohanty** [comment]
Resolved by #2035.
